stubs/jsonschema/jsonschema/_types.pyi

```diff
@@ -1,26 +1,24 @@
-from typing import Any
+from typing import Callable, Iterable, Mapping
 
-def is_array(checker, instance): ...
-def is_bool(checker, instance): ...
-def is_integer(checker, instance): ...
-def is_null(checker, instance): ...
-def is_number(checker, instance): ...
-def is_object(checker, instance): ...
-def is_string(checker, instance): ...
-def is_any(checker, instance): ...
+def is_array(checker, instance) -> bool: ...
+def is_bool(checker, instance) -> bool: ...
+def is_integer(checker, instance) -> bool: ...
+def is_null(checker, instance) -> bool: ...
+def is_number(checker, instance) -> bool: ...
+def is_object(checker, instance) -> bool: ...
+def is_string(checker, instance) -> bool: ...
+def is_any(checker, instance) -> bool: ...
 
 class TypeChecker:
-    def is_type(self, instance, type): ...
-    def redefine(self, type, fn): ...
-    def redefine_many(self, definitions=...): ...
-    def remove(self, *types): ...
-    def __init__(self, type_checkers=...) -> None: ...
-    def __lt__(self, other): ...
-    def __le__(self, other): ...
-    def __gt__(self, other): ...
-    def __ge__(self, other): ...
+    def __init__(self, type_checkers: Mapping[str, Callable[[object], bool]] = ...) -> None: ...
+    def is_type(self, instance, type: str) -> bool: ...
+    def redefine(self, type: str, fn: Callable) -> TypeChecker: ...
```

Comment:
If you don't know what arguments the function should take and what it should return, you can do this:

```suggestion
    def redefine(self, type: str, fn: Callable[...]) -> TypeChecker: ...
```